dbms/src/Core/Settings.h

```diff
@@ -200,6 +200,7 @@ struct Settings : public SettingsCollection<Settings>
     M(SettingBool, join_use_nulls, 0, "Use NULLs for non-joined rows of outer JOINs for types that can be inside Nullable. If false, use default value of corresponding columns data type.") \
     \
     M(SettingJoinStrictness, join_default_strictness, JoinStrictness::ALL, "Set default strictness in JOIN query. Possible values: empty string, 'ANY', 'ALL'. If empty, query without strictness will throw exception.") \
+    M(SettingBool, any_join_get_any_from_right_table, false, "Enable ANY RIGHT|FULL JOIN that are not consistent with LEFT one. Right table in ANY JOIN has no key copies so it leads to confusing results for RIGHT and FULL JOINs. User expects 't1 ANY LEFT JOIN t2' is equal to 't2 ANY RIGHT JOIN t1'.") \
```

Comment:
Strange setting name. If I read it verbatim, it will be the description of what ANY JOIN does (including "normal" LEFT ANY JOIN case).

Reply:
I suggest `enable_any_join_from_right_side_for_right_and_full_join` - it's even more creepy.

Reply:
We could call it ```any_join_distinct_right_table_keys``` and disable ANY INNER JOIN by default too. So when we make correct SEMI-like ANY JOINS it would enable old logic for any kind of join.